cpp/include/utilities/cython.hpp

```diff
@@ -15,53 +15,113 @@
+      case graphTypeEnum::graph_view_t_double:
+        delete graph_ptr_union.graph_view_t_double_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_float_mg:
+        delete graph_ptr_union.graph_view_t_float_mg_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_double_mg:
+        delete graph_ptr_union.graph_view_t_double_mg_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_float_transposed:
+        delete graph_ptr_union.graph_view_t_float_transposed_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_double_transposed:
+        delete graph_ptr_union.graph_view_t_double_transposed_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_float_mg_transposed:
+        delete graph_ptr_union.graph_view_t_float_mg_transposed_ptr;
+        break;
+      case graphTypeEnum::graph_view_t_double_mg_transposed:
+        delete graph_ptr_union.graph_view_t_double_mg_transposed_ptr;
+        break;
+      default: break;
+    }
+    graph_ptr_type = graphTypeEnum::null;
+  }
+
+  graphPtrUnion graph_ptr_union;
+  graphTypeEnum graph_ptr_type;
 };
 
 // Factory function for creating graph containers from basic types
-// FIXME: This should accept void* for offsets and indices as well and take a
-//        dtype directly for each instead of the enum/int.
-graph_container_t create_graph_t(raft::handle_t const& handle,
-                                 int* offsets,
-                                 int* indices,
-                                 void* weights,
-                                 int weightType,
-                                 int num_vertices,
-                                 int num_edges,
-                                 int* local_vertices,
-                                 int* local_edges,
-                                 int* local_offsets,
-                                 bool transposed,
-                                 bool multi_gpu);
+// FIXME: Should local_* values be void* as well?
+void create_graph_t(graph_container_t& graph_container,
+                    raft::handle_t const& handle,
+                    void* offsets,
+                    void* indices,
+                    void* weights,
+                    numberTypeEnum offsetType,
+                    numberTypeEnum indexType,
+                    numberTypeEnum weightType,
+                    int num_vertices,
+                    int num_edges,
+                    int* local_vertices,
+                    int* local_edges,
+                    int* local_offsets,
+                    bool transposed,
+                    bool multi_gpu);
 
 // Wrapper for calling Louvain using a graph container
 template <typename weight_t>
 weight_t call_louvain(raft::handle_t const& handle,
-                      graph_container_t graph_container,
+                      graph_container_t& graph_container,
                       int* parts,
```

Comment:
This needs to match vertex_t in the graph.  Do you want to make this a void * and cast it inside the function?

Reply:
done